src/libstd/sys_common/gnu/libbacktrace.rs

```diff
@@ -173,7 +178,7 @@ pub fn print(w: &mut Write, idx: isize, addr: *mut libc::c_void,
         for (i, &(file, line)) in fileline_buf[..fileline_count].iter().enumerate() {
             if file.is_null() { continue; } // just to be sure
             let file = unsafe { CStr::from_ptr(file).to_bytes() };
-            output_fileline(w, file, line, i == FILELINE_SIZE - 1)?;
+            output_fileline(w, file, line, i == FILELINE_SIZE - 1, format)?;
```

Comment:
I believe you should be able to accomplish item 4 in #37783 by doing something like

```rust
let mut print_line = if format == backtrace::PrintFormat::Short { false } else { true };
for (i, &(file, line)) in fileline_buf[..fileline_count].iter().enumerate() {
    if file.is_null() { continue; } // just to be sure
    let file = unsafe { CStr::from_ptr(file).to_bytes() };
    if format == backtrace::PrintFormat::Short && !print_line {
        let magic_str = "core::panicking::panic_fmt";
        if file[..magic_str.len()] == magic_str.as_bytes() {
            print_line = true
        }
        continue;
    }
    output_fileline(w, file, line, i == FILELINE_SIZE - 1, format)?;
}
```